tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -9268,7 +9308,8 @@ Sema::TemplateDeductionResult HLSLExternalSource::DeduceTemplateArgumentsForHLSL
 
   while (cursor != end)
   {
-    if (!MatchArguments(*cursor, objectElement, functionTemplateTypeArg, Args, &argTypes))
+    size_t badArg;
+    if (!MatchArguments(*cursor, objectElement, functionTemplateTypeArg, Args, &argTypes, badArg))
```

Comment:
This is a location called for methods. It has its own set of problems and executes a completely separate path to report errors. I filed #3055 to address these